From 132430dacf7d879d102f978d8a68507075a414c9 Mon Sep 17 00:00:00 2001
From: Sander <hey@sandydoo.me>
Date: Tue, 12 Aug 2025 02:48:05 +0200
Subject: [PATCH] devenv: revert dirty repo fingerprinting

Given that we allow untracked files, the fingerprinting can be extremely expensive in certain repos with large untracked datasets.
---
 src/libfetchers/git-utils.cc                  |  5 ----
 src/libfetchers/git.cc                        | 27 ++-----------------
 .../include/nix/fetchers/git-utils.hh         |  6 -----
 3 files changed, 2 insertions(+), 36 deletions(-)

diff --git a/src/libfetchers/git-utils.cc b/src/libfetchers/git-utils.cc
index 13535a161..68a3124be 100644
--- a/src/libfetchers/git-utils.cc
+++ b/src/libfetchers/git-utils.cc
@@ -421,12 +421,7 @@ struct GitRepoImpl : GitRepo, std::enable_shared_from_this<GitRepoImpl>
         {
             if (!(statusFlags & GIT_STATUS_INDEX_DELETED) &&
                 !(statusFlags & GIT_STATUS_WT_DELETED))
-            {
                 info.files.insert(CanonPath(path));
-                if (statusFlags != GIT_STATUS_CURRENT)
-                    info.dirtyFiles.insert(CanonPath(path));
-            } else
-                info.deletedFiles.insert(CanonPath(path));
             if (statusFlags != GIT_STATUS_CURRENT)
                 info.isDirty = true;
             return 0;
diff --git a/src/libfetchers/git.cc b/src/libfetchers/git.cc
index 1de9f2149..bf9fb0354 100644
--- a/src/libfetchers/git.cc
+++ b/src/libfetchers/git.cc
@@ -857,33 +857,10 @@ struct GitInputScheme : InputScheme
 
     std::optional<std::string> getFingerprint(ref<Store> store, const Input & input) const override
     {
-        auto makeFingerprint = [&](const Hash & rev)
-        {
-            return rev.gitRev() + (getSubmodulesAttr(input) ? ";s" : "") + (getExportIgnoreAttr(input) ? ";e" : "") + (getLfsAttr(input) ? ";l" : "");
-        };
-
         if (auto rev = input.getRev())
-            return makeFingerprint(*rev);
-        else {
-            auto repoInfo = getRepoInfo(input);
-            if (auto repoPath = repoInfo.getPath(); repoPath && repoInfo.workdirInfo.headRev && repoInfo.workdirInfo.submodules.empty()) {
-                /* Calculate a fingerprint that takes into account the
-                   deleted and modified/added files. */
-                HashSink hashSink{HashAlgorithm::SHA512};
-                for (auto & file : repoInfo.workdirInfo.dirtyFiles) {
-                    writeString("modified:", hashSink);
-                    writeString(file.abs(), hashSink);
-                    dumpPath((*repoPath / file.rel()).string(), hashSink);
-                }
-                for (auto & file : repoInfo.workdirInfo.deletedFiles) {
-                    writeString("deleted:", hashSink);
-                    writeString(file.abs(), hashSink);
-                }
-                return makeFingerprint(*repoInfo.workdirInfo.headRev)
-                    + ";d=" + hashSink.finish().first.to_string(HashFormat::Base16, false);
-            }
+            return rev->gitRev() + (getSubmodulesAttr(input) ? ";s" : "") + (getExportIgnoreAttr(input) ? ";e" : "") + (getLfsAttr(input) ? ";l" : "");
+        else
             return std::nullopt;
-        }
     }
 
     bool isLocked(const Input & input) const override
diff --git a/src/libfetchers/include/nix/fetchers/git-utils.hh b/src/libfetchers/include/nix/fetchers/git-utils.hh
index 2926deb4f..b8d572c0d 100644
--- a/src/libfetchers/include/nix/fetchers/git-utils.hh
+++ b/src/libfetchers/include/nix/fetchers/git-utils.hh
@@ -59,12 +59,6 @@ struct GitRepo
            modified or added, but excluding deleted files. */
         std::set<CanonPath> files;
 
-        /* All modified or added files. */
-        std::set<CanonPath> dirtyFiles;
-
-        /* The deleted files. */
-        std::set<CanonPath> deletedFiles;
-
         /* The submodules listed in .gitmodules of this workdir. */
         std::vector<Submodule> submodules;
     };
-- 
2.50.1

